src/macros.rs

```diff
@@ -12,3 +12,15 @@ macro_rules! with_traceback {
         }
     };
 }
+
+#[macro_export]
+macro_rules! log_func {
+    () => {{
+        fn f() {}
+        fn type_name_of<T>(_: T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        let name = type_name_of(f);
```

Comment:
It looks like the goal of this macro is to get the name of the containing function? How does this work?

Reply:
That is the goal although [the rust function](https://doc.rust-lang.org/std/any/fn.type_name.html) isn't as precise as I'd like.

Reply:
Candidly, the internet helped me with this, but I believe you can wrap the function in your own code and thereby access the type of the function which resolves to the fully qualified name.

Reply:
I think that makes sense. Hence the slice to `-3` to remove `f()` as the last step.